### PR TITLE
Add benchmark for macOS K/N + rendering

### PR DESCRIPTION
**Pull request by elijah-semyonov**

### CODE REVIEW

File: benchmarks/kn-performance/gradle.properties [hunk 1]
```diff
@@ -0,0 +1,17 @@
+compose.version=1.4.1
```
Comment (pjBooms) [inline benchmarks/kn-performance/gradle.properties:1]: Please change to 1.5.1 and Kotlin 1.9.10

File: benchmarks/kn-performance/src/commonMain/kotlin/Benchmarks.kt [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    stats.prettyPrint()
+    println()
+
+    return stats
+}
+
+fun runBenchmarks(
+    width: Int = 1920,
+    height: Int = 1080,
+    targetFps: Int = 60,
```
Comment (pjBooms) [inline benchmarks/kn-performance/src/commonMain/kotlin/Benchmarks.kt:126]: Please change default to 120 as we see it is more imortnat to target 120Hz devices

File: benchmarks/kn-performance/src/commonMain/kotlin/MeasureComposable.kt [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        scene.constraints = Constraints.fixed(width, height)
+        val surface = graphicsContext?.surface(width, height) ?: Surface.makeNull(width, height)
+
+        var nanoTime = 0L
+
+        val frames = MutableList(frameCount) {
+            BenchmarkFrame(Duration.INFINITE, Duration.INFINITE)
+        }
+
+        repeat(frameCount) {
```
Comment (pjBooms) [inline benchmarks/kn-performance/src/commonMain/kotlin/MeasureComposable.kt:41]: Please emulate vsync here, so we do nothing if there is a time to the next frame.

**Review by pjBooms — CHANGES_REQUESTED** [review]

File: benchmarks/kn-performance/gradle.properties [hunk 4]
```diff
@@ -0,0 +1,17 @@
+compose.version=1.5.1
+kotlin.version=1.9.10
+agp.version=7.0.4
+org.gradle.jvmargs=-Xmx3g
+kotlin.code.style=official
+kotlin.native.cacheKind=none
```
Comment (pjBooms) [inline benchmarks/kn-performance/gradle.properties:6]: remove this

Comment (pjBooms) [reply benchmarks/kn-performance/gradle.properties:6]: The following properties are mot required as well:

kotlin.native.enableDependencyPropagation=false
kotlin.mpp.enableGranularSourceSetsMetadata=true
# Enable kotlin/native experimental memory model
kotlin.native.binary.memoryModel=experimental

**Review by pjBooms — CHANGES_REQUESTED** [review]
Cleanup gradle.properties

File: benchmarks/kn-performance/gradle.properties (hunk 4 above)
Comment (elijah-semyonov) [reply benchmarks/kn-performance/gradle.properties:6]:
```
Compilation failed: Java heap space

 * Source files: 
 * Compiler version: 1.9.10
 * Output kind: PROGRAM
 ```
 When `org.gradle.jvmargs=-Xmx3g` is removed.

Comment (elijah-semyonov) [reply benchmarks/kn-performance/gradle.properties]: Removed others

**Review by pjBooms — APPROVED** [review]

**Pull request merged**